docs/reference/detr.md

```diff
@@ -0,0 +1,395 @@
+- **verbose** : *bool, default=True*
+    Enables the maximum verbosity.
+- **annotations_folder** : *str, default='Annotations'*
+    Foldername of the annotations json file. This folder should be contained in the dataset path provided.
+- **train_images_folder** : *str, default='train2017'*
+    Name of the folder that contains the train dataset images. This folder should be contained in the dataset path provided. Note that this is a folder name, not a path.
+- **train_annotations_file** : *str, default='instances_train2017.json'*
+    Filename of the train annotations json file. This file should be contained in the dataset path provided.
+-  **val_images_folder** : *str, default='val2017'*
+    Folder name that contains the validation images. This folder should be contained in the dataset path provided. Note that this is a folder name, not a path.
+- **val_annotations_file** : *str, default='instances_val2017.json'*
+    Filename of the validation annotations json file. This file should be contained in the dataset path provided in the annotations folder provided.
+
+#### `DetrLearner.eval`
+```python
+DetrLearner.eval(self, dataset, images_folder, annotations_folder, annotations_file)
+```
+
+This method is used to evaluate a trained model on an evaluation dataset.
+Returns a dictionary containing stats regarding evaluation.
+Parameters:
+- **dataset** : *object*
+    ExternalDataset class object or DatasetIterator class object. Object that holds the evaluation dataset.
+- **images_folder** : *str, default='val2017'*
+    Folder name that contains the dataset images. This folder should be contained in the dataset path provided. Note that this is a folder name, not a path.
+- **annotations_folder** : *str, default='Annotations'*
+    Folder name of the annotations json file. This file should be contained in the dataset path provided.
+- **annotations_file** : *str, default='instances_val2017.json'*
+    Filename of the annotations json file. This file should be contained in the dataset path provided.
+
+#### `DetrLearner.infer`
+```python
+DetrLearner.infer(image)
+```
+
+This method is used to perform object detection on an image.
+Returns an engine.target.BoundingBoxList object, which contains bounding boxes that are described by the left-top corner and its width and height, or returns an empty list if no detections were made.
+
+Parameters:
+- **image : *object*
+    Image of type engine.data.Image class or np.array. Image to run inference on.
+
+#### `DetrLearner.save`
+```python
+DetrLearner.save(self, path)
+```
+
+This method is used to save a trained model.
+Provided with the path, it creates the "name" directory, if it does not already
+exist. Inside this folder, the model is saved as "detr_[backbone_model].pth" and the metadata file as "detr_[backbone].json". If the directory
+already exists, the "detr_[backbone_model].pth" and "detr_[backbone].json" files are overwritten.
```

Comment:
Please write each sentence on a new line

Reply:
I will do this from now on. It is now updated.